### DISCUSSION TITLE
Hypixel Public API Changes

### DISCUSSION THREAD

**Discussion by Clemintina** [HypixelDev/PublicAPI#579] [General]
Hello all,

As stated [here](https://hypixel.net/threads/public-api-changes-february-2023.5266129/), Hypixel has chosen to make the move to remove player keys. I personally believe this is a bad idea and will not only effect players negatively but also kill any inspiration for future projects utilising the API. While thinking about this, I've had a few minor changes which would ultimately fix lots of the issues described in the article linked.

1. Caching responses for ~2 minutes
2. Vetting players who get higher rate limits more carefully
3. Adding better checks to see if the same players have been requested within a certain period from the same key
4. Using CloudFlare to cache on behalf of Hypixel to reduce bandwidth

Ultimately these minor changes would stop abuse such as Live Tracking as well as other forms of abuse which may have resulted in these changes. 

Link: https://hypixel.net/threads/public-api-changes-february-2023.5266129/

I look forward to seeing other people's ideas and preferred changes.

**Comment by mdashlw** [comment]
I'd much rather take their approach than caching. Caching should be avoided as much as possible.
